dhcpv4/client.go

```diff
@@ -140,10 +140,14 @@ func (c *Client) Exchange(ifname string, discover *DHCPv4) ([]DHCPv4, error) {
 
 	// Discover
 	if discover == nil {
-		discover, err = NewDiscoveryForInterface(ifname)
+		discover, err = NewDiscoveryForInterface(ifname, modifiers...)
```

Comment:
I wouldn't call `NewDiscoveryForInterface` with modifiers.
I would apply the modifiers once we have the discover packet.
I think we can also remove the else statement in that case.